overlord/hookstate/ctlcmd/unset.go

```diff
@@ -66,8 +68,30 @@ func (s *unsetCommand) Execute(args []string) error {
 	tr := configstate.ContextTransaction(context)
 	context.Unlock()
 
-	for _, confKey := range s.Positional.ConfKeys {
-		tr.Set(context.InstanceName(), confKey, nil)
+	if s.View {
+		if !strings.HasPrefix(s.Positional.ConfKeys[0], ":") {
+			return fmt.Errorf(i18n.G("cannot unset registry: plug must conform to format \":<plug-name>\": %s"), s.Positional.ConfKeys[0])
+		}
+
+		_, plugName, _ := strings.Cut(s.Positional.ConfKeys[0], ":")
+		if plugName == "" {
+			return fmt.Errorf(i18n.G("cannot unset registry: plug name was not provided"))
+		}
+
+		if len(s.Positional.ConfKeys) == 1 {
+			return fmt.Errorf(i18n.G("cannot unset registry: no paths provided to unset"))
```

Comment:
This error and the above could just be `errors.New` since they don't take any format args.